azure-pipelines-regression-test.yml

```diff
@@ -0,0 +1,309 @@
+
+  # Test against third-party repositories
+  - ${{ each item in parameters.testMatrix }}:
+    - job: Test_${{ replace(item.repo, '/', '_') }}
+      displayName: 'Regression Test: ${{ item.displayName }}'
+      dependsOn: BuildCompiler
+      pool:
+        name: $(DncEngPublicBuildPool)
+        demands: ImageOverride -equals windows.vs2022.amd64.open
+      timeoutInMinutes: 60
+      variables:
+        TestRepoName: ${{ item.repo }}
+        TestCommit: ${{ item.commit }}
+        BuildScript: ${{ item.buildScript }}
+        DisplayName: ${{ item.displayName }}
+      steps:
+      - checkout: none
+        displayName: Skip default checkout
+
+      # Download the F# compiler artifacts
+      - task: DownloadPipelineArtifact@2
+        displayName: Download F# Compiler Artifacts
+        inputs:
+          artifactName: 'FSharpCompilerArtifacts'
+          downloadPath: '$(Pipeline.Workspace)/FSharpCompiler'
+
+      - task: DownloadPipelineArtifact@2
+        displayName: Download UseLocalCompiler props
+        inputs:
+          artifactName: 'UseLocalCompilerProps'
+          downloadPath: '$(Pipeline.Workspace)/Props'
+
+      # Checkout the third-party repository at specific commit
+      - script: |
+          echo Cloning repository: $(TestRepoName)
+          git clone https://github.com/$(TestRepoName).git $(Pipeline.Workspace)/TestRepo
+          cd $(Pipeline.Workspace)/TestRepo
+          
+          echo Checking out commit: $(TestCommit)
+          git checkout $(TestCommit)
+          
+          echo Successfully checked out $(TestRepoName) at commit $(TestCommit)
+          git log -1 --oneline
+          
+          echo Repository structure:
+          dir /B
+          
+          echo Verifying build script exists: $(BuildScript)
+          if exist "$(BuildScript)" (
+            echo ✓ Build script found: $(BuildScript)
+          ) else (
+            echo ✗ Build script not found: $(BuildScript)
+            echo Available files in root:
+            dir
+            exit /b 1
+          )
+        displayName: 'Checkout $(DisplayName) at specific commit'
+
+      # Inject UseLocalCompiler.Directory.Build.props
+      - script: |
+          cd $(Pipeline.Workspace)/TestRepo
+          echo Creating Directory.Build.props to use local F# compiler
+          echo ^<Project^> > Directory.Build.props
+          echo   ^<Import Project="UseLocalCompiler.Directory.Build.props" /^> >> Directory.Build.props
+          echo ^</Project^> >> Directory.Build.props
+          
+          echo Copying UseLocalCompiler.Directory.Build.props with updated path
+          powershell -Command "& { $content = Get-Content '$(Pipeline.Workspace)/Props/UseLocalCompiler.Directory.Build.props' -Raw; $content = $content -replace 'LocalFSharpCompilerPath.*MSBuildThisFileDirectory.*', 'LocalFSharpCompilerPath>$(Pipeline.Workspace)/FSharpCompiler<'; Set-Content 'UseLocalCompiler.Directory.Build.props' -Value $content }"
```

Comment:
Msbuild can read the `LocalFSharpCompilerPath` from an env var, which you can just set at the azdo job level.
WHen you do that, you can just refer to the `UseLocalCompiler.Directory.Build.props` as is, no copying needed.